gmso/formats/mcf.py

```diff
@@ -0,0 +1,700 @@
+            )
+        )
+
+        mcf.write(header)
+        _write_atom_information(mcf, top, in_ring)
+        _write_bond_information(mcf, top)
+        _write_angle_information(mcf, top)
+        _write_dihedral_information(mcf, top)
+        # TODO: Add improper information
+        # _write_improper_information(mcf, top)
+        _write_fragment_information(mcf, top, frag_list, frag_conn)
+        _write_intrascaling_information(mcf, lj14, coul14)
+
+        # That's all, folks!
+        mcf.write("\n\nEND\n")
+
+
+def _id_rings_fragments(top):
+    """Identifies the rings and fragments of the molecule
+
+    Parameters
+    ----------
+    top : gmso.core.Topology
+        Topology object
+
+    Returns
+    ---------
+    in_ring : list
+        True for each atom in a ring
+    frag_list : list
+        Atom ids belonging to each fragment
+    frag_conn : list
+        Fragment ids of connected fragments
+
+    """
+
+    # Identify atoms in rings
+    bond_graph = nx.Graph()
+    bond_graph.add_edges_from(
+        [[bond.atom1.idx, bond.atom2.idx] for bond in top.bonds]
+    )
+    if len(top.bonds) == 0:
+        warnings.warn(
+            "No bonds found. Cassandra will interpet "
+            "this as a rigid species"
+        )
+        in_ring = [False] * len(top.sites)
+        frag_list = []
+        frag_conn = []
+        return in_ring, frag_list, frag_conn
+
+    # Check if entire molecule is connected. Warn if not.
+    if nx.is_connected(bond_graph) == False:
+        raise ValueError(
+            "Not all components of the molecule are connected. "
+            "MCF files are for a single molecule and thus "
+            "everything should be connected through bonds."
+        )
+
+    all_rings = nx.cycle_basis(bond_graph)
+    in_ring = [False] * bond_graph.number_of_nodes()
+    adj_to_ring = [False] * bond_graph.number_of_nodes()
+    for ring in all_rings:
+        for idx in ring:
+            in_ring[idx] = True
+
+    # Identify fragments
+    # See Shah and Maginn, JCP, 135, 134121, 2011, doi:10.1063/1.3644939
+    frag_list = []
+    frag_conn = []
+
+    # First create a neighbor list for each atom
+    neigh_dict = {
+        i: list(bond_graph.neighbors(i))
+        for i in range(bond_graph.number_of_nodes())
+    }
+    # First ID fused rings
+    fused_rings = []
+    rings_to_remove = []
+    for i in range(len(all_rings)):
+        ring1 = all_rings[i]
+        for j in range(i + 1, len(all_rings)):
+            ring2 = all_rings[j]
+            shared_atoms = list(set(ring1) & set(ring2))
+            if len(shared_atoms) == 2:
+                fused_rings.append(list(set(ring1 + ring2)))
+                rings_to_remove.append(ring1)
+                rings_to_remove.append(ring2)
+    for ring in rings_to_remove:
+        all_rings.remove(ring)
+    all_rings = all_rings + fused_rings
+    # ID fragments which contain a ring
+    for ring in all_rings:
+        adjacentatoms = []
```

Comment:
```suggestion
        adjacent_atoms = []
```

Makes it a tad easier to read.